core/vm/lfvm/instructions.go

```diff
@@ -511,16 +620,35 @@ func opGasPrice(c *context) {
 func opBalance(c *context) {
 	slot := c.stack.peek()
 	address := common.Address(slot.Bytes20())
+	err := gasEip2929AccountCheck(c, address)
+	if err != nil {
+		c.status = OUT_OF_GAS
```

Comment:
nit: OUT_OF_GAS state could be set `gasEip2929AccountCheck`

Reply:
done